### ISSUE TITLE
IPTB plugin to connect IPTB tests to js-libp2p and go-libp2p daemons

### ISSUE THREAD

**Issue by ghost** [libp2p/libp2p#46]
@jhiesey (stretch goal for end of Sept)

- Create js-libp2p daemon if one does not already exist
- IPTB plugin to connect IPTB tests to js-libp2p and go-libp2p daemons

/ cc @jacobheun - John should probably sync with you so he doesn't reinvent any wheels here

Ref: https://docs.google.com/document/d/1XOmU-9V6yx0tMcW9iVeL9nNSgrPDt5X7aBmRzD2ZO34/edit

**Comment by jacobheun** [comment]
I haven't started on the daemon work yet, as we still need to come to a decision on a basic api with the rest of the libp2p team unless that was done during dht day. I will be working on state machine updates to libp2p in the next week, once i finish that work on libp2p-switch, but that shouldn't conflict and should hopefully make future daemon work easier.

**Comment by ghost** [comment]
No, DHT Day did not come up with a daemon API, just a consensus that we needed one.

Is there an issue or any doc with previous discussion on daemon API that we could point @jhiesey and @bigs toward to further the discussion?

**Comment by jacobheun** [comment]
There's this discussion that started last year around the daemon, https://github.com/libp2p/libp2p/issues/22.  The focus for this issue is around the dht so we probably just need to focus on the core functionality needed to interact with the dht. #22 is probably the place to have that discussion.
